components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,1275 @@
+        BookHebrew::is_hebrew_leap_year(civil_year)
+    }
+
+    fn last_month_day_in_year(civil_year: i32) -> (u8, u8) {
+        let civil_month = Self::last_month_of_civil_hebrew_year(civil_year);
+        let civil_day = Self::last_day_of_civil_hebrew_month(civil_year, civil_month);
+
+        (civil_month, civil_day)
+    }
+}
+
+impl Calendar for Hebrew {
+    type DateInner = HebrewDateInner;
+
+    fn date_from_codes(
+        &self,
+        _era: types::Era,
+        _year: i32,
+        _month_code: types::MonthCode,
+        _day: u8,
+    ) -> Result<Self::DateInner, CalendarError> {
+        // let year = if era.0 == tinystr!(16, "hebrew") {
+        //     year
+        // } else {
+        //     return Err(CalendarError::UnknownEra(era.0, self.debug_name()));
+        // };
+
+        // ArithmeticDate::new_from_codes(self, year, month_code, day).map(HebrewDateInner)
+        todo!("#3789")
+    }
+
+    fn date_from_iso(&self, iso: Date<Iso>) -> Self::DateInner {
+        let fixed_iso = Iso::fixed_from_iso(*iso.inner());
+        Self::civil_hebrew_from_fixed(fixed_iso).inner
+    }
+
+    fn date_to_iso(&self, date: &Self::DateInner) -> Date<Iso> {
+        let fixed_hebrew = Self::fixed_from_civil_hebrew(*date);
+        Iso::iso_from_fixed(fixed_hebrew)
+    }
+
+    fn months_in_year(&self, date: &Self::DateInner) -> u8 {
+        date.0.months_in_year()
+    }
+
+    fn days_in_year(&self, date: &Self::DateInner) -> u16 {
+        date.0.days_in_year()
+    }
+
+    fn days_in_month(&self, date: &Self::DateInner) -> u8 {
+        date.0.days_in_month()
+    }
+
+    fn offset_date(&self, date: &mut Self::DateInner, offset: DateDuration<Self>) {
+        date.0.offset_date(offset)
+    }
+
+    fn until(
+        &self,
+        date1: &Self::DateInner,
+        date2: &Self::DateInner,
+        _calendar2: &Self,
+        _largest_unit: DateDurationUnit,
+        _smallest_unit: DateDurationUnit,
+    ) -> DateDuration<Self> {
+        date1.0.until(date2.0, _largest_unit, _smallest_unit)
+    }
+
+    fn debug_name(&self) -> &'static str {
+        "Hebrew"
+    }
+
+    fn year(&self, date: &Self::DateInner) -> types::FormattableYear {
+        Self::year_as_hebrew(date.0.year)
+    }
+
+    fn month(&self, date: &Self::DateInner) -> types::FormattableMonth {
+        let mut ordinal = date.0.month;
+        let is_leap_year = Self::is_leap_year(date.0.year);
+
+        if is_leap_year && ordinal == 6 {
+            return types::FormattableMonth {
+                ordinal: ordinal as u32,
+                code: types::MonthCode(tinystr!(4, "M05L")),
+            };
+        }
+
+        if is_leap_year && ordinal > 6 {
+            ordinal -= 1;
+        }
+
+        let code = match ordinal {
+            1 => tinystr!(4, "M01"),
+            2 => tinystr!(4, "M02"),
+            3 => tinystr!(4, "M03"),
+            4 => tinystr!(4, "M04"),
+            5 => tinystr!(4, "M05"),
+            6 => tinystr!(4, "M06"),
```

Comment:
With this code, both `M05L` and `M06` have an ordinal of `6`. Is that correct?

Reply:
Yeah, the ordinal month number is a bit strange; it's only used for formatting and only when there is a `M` or `MM` in the pattern. In Hebrew length and skeleton patterns there is only `MMM`, so the ordinal is not shown to users. It's simply not clear what `ordinal` is supposed to be.

https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-cal-hebrew-full/main/en/ca-hebrew.json#L324

For ECMAScript Temporal purposes, the month number has a clear meaning, but `FormattableMonth` should not be used for that; see https://github.com/unicode-org/icu4x/issues/3821

Reply:
In general we may even wish to soft error on CLDR calendar files which use M/MM patterns for calendars with leap months unless they also provide numeric formatting patterns for them (which is what Chinese does)